sound/soc/sof/intel/hda-loader.c

```diff
@@ -149,8 +150,12 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, const void *fwdata,
 					HDA_DSP_REG_POLL_INTERVAL_US,
 					chip->rom_init_timeout *
 					USEC_PER_MSEC);
-	if (!ret)
-		return 0;
+	if (ret < 0) {
+		dev_err(sdev->dev,
+			"error: %s: timeout HDA_DSP_SRAM_REG_ROM_STATUS read\n",
+			__func__);
+		return ret;
+	}
```

Comment:
@plbossart The massive failures in CI are probably because of this,  you should "return 0;" here. Now it goes to passes through to "err: " and it powers down the core.

Reply:
@kv2019i I am not sure I understand your point. did you mean 
a) return 0 at line 157? If yes, why?
b) add a return 0 before the err: label? if yes, why?

Reply:
@plbossart In the original code, in success case, if statement on L152 is taken and function returns 0 on L153.
 
In your patch, on success case "if (ret < 0)" is not taken I code continues to L158-160 and to code under "err:" label.

Reply:
@kv2019i well, I question the original code really. What is the point of removing the error here? It will prevent the loop on ROM inits from happening. It's just bad.

@keyonjie @ranj063 can you please comment here. I don't have the history but without comments it looks like a mistake, not a feature.

Reply:
@plbossart Aa, but we are not muting an error in original code. "if (!ret)" is the success case, you've removed that one and left only error cases in code, so the function will always fail.

Reply:
Gah, yes you are right.
That's precisely why I don't like it when people use non-regular filters. It's just too easy to screw up.
Will fix this.